packages/eslint-plugin/docs/rules/no-unsafe-type-assertion.mdx

```diff
@@ -0,0 +1,61 @@
+---
+description: 'Disallow type assertions that widen a type.'
+---
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+> 🛑 This file is source code, not the primary documentation location! 🛑
+>
+> See **https://typescript-eslint.io/rules/no-unsafe-type-assertion** for documentation.
+
+This rule forbids using type assertions to narrow a type, as it can lead to unsafe assumptions that bypass TypeScript’s type-checking.
+
+Instead, it's better to rely on type guards to ensure type safety and avoid potential runtime errors caused by unsafe type assertions.
+
+## Examples
+
+<Tabs>
+<TabItem value="❌ Incorrect">
+
+```ts
+function f() {
+  return Math.random() < 0.5 ? 42 : 'oops';
+}
+
+const z = f() as number;
+
+const items = [1, '2', 3, '4'];
+
+const number = items[0] as number;
+```
+
+</TabItem>
+<TabItem value="✅ Correct">
+
+```ts
+function f() {
+  return Math.random() < 0.5 ? 42 : 'oops';
+}
+
+const z = f() as number | string | boolean;
+
+const items = [1, '2', 3, '4'];
+
+const number = items[0] as number | string | undefined;
+```
+
+</TabItem>
+</Tabs>
+
+## When Not To Use It
+
+If your codebase has many unsafe type assertions, then it may be difficult to enable this rule.
+It may be easier to skip the `no-unsafe-*` rules pending increasing type safety in unsafe areas of your project.
+You might consider using [ESLint disable comments](https://eslint.org/docs/latest/use/configure/rules#using-configuration-comments-1) for those specific situations instead of completely disabling this rule.
+
+This rule requires TypeScript v5.4 or above; you should not use it if your TypeScript version is lower.
```

Comment:
This isn't correct - the `isTypeAssignableTo` method is only _publicly documented_ on the API starting in v5.4, but it's available at runtime in all our supported versions. Indeed, that's why we've waited until now to use it! 🙂 

The public visibility of the `isTypeAssignableTo` is nothing but an implementation detail on our end, and not something that affects users, so this adominition about the rule being only available in TS v5.4 and up should be removed. Feel free to [try the rule in the playground using our oldest supported TS version](https://deploy-preview-10051--typescript-eslint.netlify.app/play/#ts=4.8.4&fileType=.ts&code=CYUwxgNghgTiAEYD2A7AzgF3gDwFz0xgEsUBzeAH3hQFcBbAIxBgG4AoN7eKNAjYsiyA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1tieQEMAZolpFStPsmSJo%2BShybooAdz7QF4MAF8QmoA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eFYDAruuGAL4g9A&tokens=false) 🙂